File: 2099-number-of-strings-that-appear-as-substrings-in-word/number-of-strings-that-appear-as-substrings-in-word.cpp
class Solution {
public:
    int numOfStrings(vector<string>& patterns, string word) {
        unordered_map<string,int> MAP;
        int n = word.size();
        for (int i = 0; i<n; i++){
            string S;
            for (int j = i;j<n;j++){S+=word[j]; MAP[S]++;}
        }
        int c = 0;
        for (string &w: patterns){c += MAP[w]>=1;}
        return c;
    }
};

File: 2099-number-of-strings-that-appear-as-substrings-in-word/number-of-strings-that-appear-as-substrings-in-word.py
class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        MAP = defaultdict(int)
        n = len(word)
        for i in range(n):
            currs = ""
            for j in range(i,n):
                currs+=word[j]
                MAP[currs]+=1
        count = 0
        for word in patterns:
            count += MAP[word]>=1
        return count